docs/examples/getDataMoverResources.go

```diff
@@ -0,0 +1,315 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	backup  bool
+	restore bool
+	details bool
+
+	kubernetesClient *dynamic.DynamicClient
+
+	VolumeSnapshotResource = schema.GroupVersionResource{
+		Group:    "snapshot.storage.k8s.io",
+		Version:  "v1",
+		Resource: "volumesnapshots",
+	}
+	VolumeSnapshotContentResource = schema.GroupVersionResource{
+		Group:    "snapshot.storage.k8s.io",
+		Version:  "v1",
+		Resource: "volumesnapshotcontents",
+	}
+	VolumeSnapshotBackupResource = schema.GroupVersionResource{
+		Group:    "datamover.oadp.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "volumesnapshotbackups",
+	}
+	ReplicationSourceResource = schema.GroupVersionResource{
+		Group:    "volsync.backube",
+		Version:  "v1alpha1",
+		Resource: "replicationsources",
+	}
+)
+
+var rootCmd = &cobra.Command{
+	Use: "go run docs/examples/getDataMoverResources.go",
+	Long: `Check the OADP DataMover Custom Resources in real time.
+
+You need to log in to your cluster and deploy OADP prior to run this script`,
+	Args: cobra.NoArgs,
+	// TODO add examples
+	Example: "  go run docs/examples/getDataMoverResources.go -b",
+	Run:     rootRun,
+}
+
+// TODO transform into flag?
+const namespace = "openshift-adp"
+
+// no --tty flag: Unable to use a TTY - input is not a terminal or the right kind of file
+const velero = "oc -n %v exec deployment/velero -c velero -i -- ./velero %v"
+
+func exitOnError(err error) {
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func getKubernetesClient() {
+	kubernetesConfig := filepath.Join(homedir.HomeDir(), ".kube", "config")
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubernetesConfig)
+	exitOnError(err)
+
+	kubernetesClient, err = dynamic.NewForConfig(config)
+	exitOnError(err)
+}
+
+func getKubernetesResourceList(resourceSpecification schema.GroupVersionResource, namespace string) *unstructured.UnstructuredList {
+	resource, err := kubernetesClient.Resource(resourceSpecification).Namespace(namespace).List(context.TODO(), metav1.ListOptions{})
+	exitOnError(err)
+	return resource
+}
+
+func backupSummaryKubeLib(details bool) {
```

Comment:
tested @mpryc suggestion. I liked :smile:  what you prefer so I can apply to the remaining: shell or Kubernetes lib?

Reply:
go lib would be more reliable but adds go dependency that could come back to haunt us.

Don't go depend on something not meant for consumption like [k8s.io/kubectl/pkg/cmd/cp](https://gitlab.cee.redhat.com/app-mig/oadp-e2e-qe/-/commit/1281a66f866bf39b2a85b450302af5f7c7e1c374#a1f4f20d8f21da3825a49a53c8146f0230a76a67_39_36) tho :P

Reply:
yeah, that is a valid concern. Even without Kubernetes lib, we are using cobra, which is not direct as well

since Go does not have development dependencies, I think the best solution would be a separate repo and use [Makefile go-install-tool](https://github.com/openshift/oadp-operator/blob/master/Makefile#L284-L296) (which can become 2 repos changes to achieve a simple change in the future). Do you suggest another solution to overcome Go dependency problem?

Reply:
I mean you could still consume kubectl via go like in https://gitlab.cee.redhat.com/app-mig/oadp-e2e-qe/-/blob/master/utils/pods/kube_exec.go#L37

Just consume the top main cobra (kubectl) command, rather than digging into subcommand for example.

I'm not as worried about client-go lib as we import that already.

Reply:
I think I still did not understand your suggestion @kaovilai 

do you suggest to have a function like this?
```go
func kubectl(args ...string) string {
  kubectl := cmd.NewDefaultKubectlCommand()
  kubectl.SetArgs(args...)
  outputBuffer := new(bytes.Buffer)
  kubectl.SetOut(outputBuffer)
  err := kubectl.Execute()
  exitOnError(err)
  return outputBuffer.String()
}

kubectl([]string{"get", "volumesnapshot", "-A"})
```

Reply:
Correct.. if you are to do something with kubectl, that would be easier/faster to consume and reliable (shell independent).

Reply:
if you can use client-go instead, that would be preferable but otherwise it's fine.
https://stackoverflow.com/questions/43314689/example-of-exec-in-k8ss-pod-by-using-go-client

Reply:
oc get volumesnapshots -A should definitely use client-go, you can call length on the result easily.

Reply:
alright, thanks! going to study it and make the required changes

Reply:
used go client for everything :) resolving conversation, but feel free to open it again if you see any problems